compiler/src/iree/compiler/Codegen/Common/GPU/test/gpu_apply_tiling_level.mlir

```diff
@@ -548,15 +548,41 @@ func.func @partial_reduction(%3: tensor<?x?xf32>) -> tensor<?xf32> {
 
 // -----
 
+#config = #iree_gpu.lowering_config<{reduction = [0, 1]}>
+func.func @swap_collapse_shape_with_extract_slice_block_arg(%arg0: tensor<16x1x1x16xf32>) -> tensor<16x16xf32> {
```

Comment:
Where does the "block arg" come from? is it a result of the config values?

Reply:
This is a two-step process:
1. Apply tiling based on the lowering config which generates:
```
%collapsed = tensor.collapse_shape %arg0 [[0], [1, 2, 3]] : tensor<16x1x1x16xf32> into tensor<16x16xf32>
 %1 = scf.for %arg1 = %c0 to %c16 step %c1 iter_args(%arg2 = %0) -> (tensor<16x16xf32>) {
    %extracted_slice = tensor.extract_slice %collapsed[0, %arg1] [16, 1] [1, 1] : tensor<16x16xf32> to tensor<16x1xf32>
}
```
2. Swap `collapse_shape` with `extract_slice` op. Here the collapsed offset `%arg1` is a block argument.